Documentation/class-libraries.md

```diff
@@ -0,0 +1,46 @@
+.NET Class Libraries
+====================
+
+Class libraries are the [shared library](http://en.wikipedia.org/wiki/Library_(computing)#Shared_libraries) concept for .NET. They enable you to componentize useful functionality into modules that can be used by multiple applications. They can also be used as a means of loading functionality that is not needed or not known at application startup. 
+
+There are three types of class libraries that you can use:
+
+- **Platform-specific** class libraries have access to all the APIs in a given platform (e.g. .NET Framework 4.6, Xamarin iOS), but can only be used by apps and libraries that target that platform.
+- **Portable** class libraries have access to a subset of APIs, and can be used by apps and libraries that target multiple platforms.
+- **.NET Core** class libraries are a merger of the platform-specific and portable library concept into a single model that provides the best of both.
+
+Platform-specific Class Libraries
+=================================
+
+Platform-specific libraries are bound to a single .NET platform (e.g. .NET Framework 4.6 on Windows) and can therefore take significant dependencies on a known execution environment. Such an environment will expose a known set of APIs (.NET and OS APIs) and will maintain and expose expected state (e.g. Windows registry). 
```

Comment:
I think we need some section that details the relationship with Mono. Mono ships the "full" .NET Framework (or at least a good chunk of it) for many platforms, so people new to the ecosystem will be confused when you refer to it as platform-specific (as Mono has traditionally been thought of the cross-platform .NET).